core/sdk/src/main/kotlin/com/viam/sdk/core/component/encoder/Encoder.kt

```diff
@@ -0,0 +1,135 @@
+     *             position the encoder returns (ticks or degrees).
+     *
+     */
+    abstract fun getPosition(positionType: PositionType?, extra: Struct): Pair<Float, PositionType>
+
+    /**
+     * Report the position of the encoder.
+     * The value returned is the current position in terms of its ``position_type``.
+     * If no position type is given, the supported position type will be returned.
+     * The position will be either in relative units (ticks away from a zero position) for
+     * ``PositionType.POSITION_TYPE_TICKS_COUNT`` or absolute units (degrees along a circle)
+     * for ``PositionType.POSITION_TYPE_ANGLE_DEGREES``.
+     * @return A tuple containing two values; the first [0] the position of the encoder which can either be
+     *             ticks since last zeroing for a relative encoder or degrees for an absolute encoder, and the second [1] the type of
+     *             position the encoder returns (ticks or degrees).
+     */
+    fun getPosition(positionType: PositionType): Pair<Float, PositionType> {
+        return getPosition(positionType, Struct.getDefaultInstance())
+    }
+
+    /**
+     * Report the position of the encoder.
+     * The value returned is the current position in terms of its ``position_type``.
+     * If no position type is given, the supported position type will be returned.
+     * The position will be either in relative units (ticks away from a zero position) for
+     * ``PositionType.POSITION_TYPE_TICKS_COUNT`` or absolute units (degrees along a circle)
+     * for ``PositionType.POSITION_TYPE_ANGLE_DEGREES``.
+     * @return A tuple containing two values; the first [0] the position of the encoder which can either be
+     *             ticks since last zeroing for a relative encoder or degrees for an absolute encoder, and the second [1] the type of
+     *             position the encoder returns (ticks or degrees).
+     *
+     */
+    fun getPosition(extra: Struct): Pair<Float, PositionType> {
+        return getPosition(null, extra)
+    }
+
+    /**
+     * Report the position of the encoder.
+     * The value returned is the current position in terms of its ``position_type``.
+     * If no position type is given, the supported position type will be returned.
+     * The position will be either in relative units (ticks away from a zero position) for
+     * ``PositionType.POSITION_TYPE_TICKS_COUNT`` or absolute units (degrees along a circle)
+     * for ``PositionType.POSITION_TYPE_ANGLE_DEGREES``.
+     * @return A tuple containing two values; the first [0] the position of the encoder which can either be
+     *             ticks since last zeroing for a relative encoder or degrees for an absolute encoder, and the second [1] the type of
+     *             position the encoder returns (ticks or degrees).
+     */
+    fun getPosition(): Pair<Float, PositionType> {
```

Comment:
Why does `getPosition` have four definitions? It seems other functions have two.